knowledge_graphs/knowledge_graph_validator_refactored.py

```
@@ -51,7 +46,7 @@ class MethodValidation:
     validation: ValidationResult
     expected_params: list[str] = field(default_factory=list)
     actual_params: list[str] = field(default_factory=list)
-    parameter_validation: ValidationResult = None
+    parameter_validation: Optional[ValidationResult] = None
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)